typings/connection.d.ts

```diff
@@ -0,0 +1,544 @@
+  /**
+   * @property {number} [ttl] time to live in ms.
+   */
+  ttl?: number;
+  /**
+   * @property {boolean} [durable] Specifies durability requirements.
+   */
+  durable?: boolean;
+  /**
+   * @property {number} [priority] The relative message priority. Higher numbers indicate higher
+   * priority messages.
+   */
+  priority?: number;
+  /**
+   * @property {string} [subject] A common field for summary information about the message
+   * content and purpose.
+   */
+  subject?: string;
+  /**
+   * @property {string} [user_id] The identity of the user responsible for producing the message.
+   */
+  user_id?: string;
+
+
+}
+
+/**
+ * Describes the AMQP message that is sent or received on the wire.
+ * @interface AmqpMessage
+ */
+export interface AmqpMessage extends AmqpMessageProperties {
+  // TODO: Ask Gordon about other AMQP message properties like durable, first_acquirer, etc.
+  // https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-amqp-protocol-guide#messages
+  body: any;
+  message_annotations?: AmqpMessageAnnotations;
+  application_properties?: Dictionary<any>;
+  delivery_annotations?: DeliveryAnnotations;
+}
+
+/**
+ * Defines the AMQP Connection context. This context is provided when you add an
+ * event handler to any of the objects created by rhea.
+ * @interface Context
+ */
+export interface Context {
+  /**
+   * @property {Connection} connection The amqp connection.
+   */
+  connection: Connection;
+  /**
+   * @property {Container} container The amqp container
+   */
+  container: Container;
+  /**
+   * @property {Delivery} [delivery] The amqp delivery that is received after sending a message.
+   */
+  delivery?: Delivery;
+  /**
+   * @property {AmqpMessage} [message] The amqp message that is received in the message event
+   * handler when rhea emits a message event on a receiver.
+   */
+  message?: AmqpMessage;
+  /**
+   * @property {Receiver} [receiver] The amqp receiver link that was created on the amqp connection.
+   */
+  receiver?: Receiver;
+  /**
+   * @property {Session} session The amqp session link that was created on the amqp connection.
+   */
+  session: Session;
+  /**
+   * @property {Sender} [sender] The amqp sender link that was created on the amqp connection.
+   */
+  sender?: Sender;
+}
+
+/**
+ * Defines the amqp error object.
+ * @interface AmqpError
+ */
+export interface AmqpError {
+  /**
+   * @property {string} [condition] Describes the error condition.
+   */
+  condition?: string;
+  /**
+   * @property {string} [description] Describes any supplementary information that is not indicated the error condition.
+   */
+  description?: string;
+  /**
+   * @property {any} [info] Describes the information about the error condition.
+   */
+  info?: any;
+  /**
+   * @property {any[]} [value] Describes the associated amqp value types.
+   */
+  value?: any[];
+}
+
+export declare interface Connection extends EventEmitter {
+  [x: string]: any;
```

Comment:
Yes it does. I added it initially because I wasn't sure what were the other properties on the Connection object. I can definitely remove it, once we are sure that we have covered what we want.